src/utils/variableType.js

```diff
@@ -7,10 +8,28 @@ import {isFunction} from './functionType';
  * @return {Boolean}
  */
 export function isUpdate(node) {
-  return node.type === 'UpdateExpression' && node.argument.type === 'Identifier';
+  return node.type === 'UpdateExpression';
 }
 
 /**
+ * True when node is an assignment with the variable on the LHS (like x= foo).
+ *
+ * @param {Object} node
+ * @param {Object} parent Immediate parent node (to determine context)
+ * @return {Boolean}
+ */
+export function isOnLHSAssignment(node, parent) {
+  return (
+    (
+      parent.type === 'ExpressionStatement' ||
+      parent.type === 'AssignmentExpression'
+    ) &&
+    isEqualAst(parent.left, node)
```

Comment:
This doesn't look right: `ExpressionStatement` does not have a `left` property.

Also, I think you really should be doing identity check in here: `parent.left === node`